### DISCUSSION TITLE
Argument #1 ($domaine) must be of type App\Entity\Domaine, string given

### DISCUSSION THREAD

**Discussion by mednoor890** [symfony/symfony#46861] [Q&A]
i m trying to make a post request so i can add a client which is an entity,inside this entity i have to add a relationship ,a oneToMany one
like the below code snippet shows
`#[ORM\OneToMany(mappedBy: 'clientProp', targetEntity: Domaine::class, orphanRemoval: true)]
    private $domaines;`
untill now everything is ok ,However i want to send the request from reactJs so i m wondering if i can access the field domaineName of the entity domain (related to client) inside my controller or is there any other solution to make react understand the domain entity type which i think it s not convenient.
Here s the line where the error is occuring inside my controller
`$client->addDomaine(($request->request->get("domaine")));`

**Comment by stof** [comment]
`$request->request->get("domaine")` is likely to be a string, not a Domaine object. there is no way for the Request object to contain a Domaine entity directly. You need to handle some conversion logic (either creating a new Domaine object or fetching it from the DB, depending on the use case)

**Reply by mednoor890** [reply]
actually i tried this,but when i try to add a domain which is related to the client (ManyToOne)i get this error 
**An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'client_prop_id' cannot be null** it s clearly because of the relation here s my code 
 `$domain->setClientProp($request->request->get("clientPropriet"));` adding it from a reactjs interface by the help of a select menu and values fetched from client 
and here s the code  of the concerned line of domain entity 
`#[ORM\ManyToOne(targetEntity: Client::class, inversedBy: 'domaines')]
    #[ORM\JoinColumn(nullable: false)]
    private $clientProp;`  
I didn t add the id because if i m not mistaken when we work with twig we add just the object and doctrine will fill the id automatically ,unfortunately that s not my case.
